#! /bin/sh -e

# DP: gpc: Declare global variable version string

dir=
if [ $# -eq 3 -a "$2" = '-d' ]; then
    pdir="-d $3"
    dir="$3/"
elif [ $# -ne 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
    -patch)
        patch $pdir -f --no-backup-if-mismatch -p0 < $0
        ;;
    -unpatch)
        patch $pdir -f --no-backup-if-mismatch -R -p0 < $0
        ;;
    *)
        echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
        exit 1
esac
exit 0

--- gcc/p/options.c.orig	Sun Apr 10 01:23:17 2005
+++ gcc/p/options.c	Sun Apr 10 01:26:37 2005
@@ -31,6 +31,8 @@
 #include "options.h"
 #endif
 
+const char *lang_version_string = GPC_RELEASE_STRING ", based on gcc-";
+
 /* An array with all long Pascal command-line switches. */
 static const struct
 {
@@ -915,8 +917,6 @@
       version_flag = 1;
   if (version_flag)
     fprintf (stderr, "GNU Pascal version is actually %s, based on gcc-%s\n", GPC_RELEASE_STRING, version_string);
-#else
-  version_string = concat (GPC_RELEASE_STRING ", based on gcc-", version_string, NULL);
 #endif
 
   lexer_options = compiler_options = co = (struct options *) xmalloc (sizeof (struct options));

